app/mempool/lane.go

```diff
@@ -31,6 +31,11 @@ type Lane struct {
 	// txIndex holds the uppercase hex-encoded hash for every transaction
 	// currently in this lane's mempool.
 	txIndex map[string]struct{}
+
+	// OnFilled is a callback function that is called when the lane exceeds its limit.
+	OnFilled func(isFilled bool)
```

Comment:
please dont make this public, keep it as onFilledFn and create a function `OnFilled()` that calls this internally. See